core/domain/wipeout_service.py

```diff
@@ -841,26 +906,27 @@ def _pseudonymize_models_transactional(
         for metadata_model in metadata_models:
             metadata_model.committer_id = pseudonymized_id
             metadata_model.update_timestamps()
+        datastore_services.put_multi(metadata_models)
 
         commit_log_models = [
             model for model in activity_related_models
             if isinstance(model, commit_log_model_class)]
         for commit_log_model in commit_log_models:
             commit_log_model.user_id = pseudonymized_id
             commit_log_model.update_timestamps()
-        datastore_services.put_multi(metadata_models + commit_log_models)
+        datastore_services.put_multi(commit_log_models)
 
     activity_ids_to_pids = (
         pending_deletion_request.pseudonymizable_entity_mappings[
             activity_category.value])
     for activity_id, pseudonymized_id in activity_ids_to_pids.items():
-        activity_related_models = [
+        activity_related_models: List[base_models.BaseModel] = [
             model for model in snapshot_metadata_models
             if model.get_unversioned_instance_id() == activity_id
-        ] + [
-            model for model in commit_log_models
-            if getattr(model, commit_log_model_field_name) == activity_id
         ]
+        for model in commit_log_models:
+            if getattr(model, commit_log_model_field_name) == activity_id:
+                activity_related_models.append(model)
```

Comment:
Similar to the above, we are adding two different types of list comprehensions here.